examples/kalman_filter.py

```diff
@@ -0,0 +1,374 @@
+                observer.update(obs[(i // obs_freq)])
+            traj.append(1 * observer.state)
+            uncertainty.append(1 * gs.diagonal(observer.covariance))
+        traj = gs.array(traj)
+        uncertainty = gs.array(uncertainty)
+        three_sigmas = 3 * gs.sqrt(uncertainty)
+
+        return traj, three_sigmas
+
+    np.random.seed(12345)
+    model = LocalizationLinear()
+    kalman = KalmanFilter(model)
+    n_traj = 1000
+    obs_freq = 50
+    dt = 0.1
+    P0 = gs.array([10., 1.])
+    P0 = np.diag(P0)
+    Q = 0.001 * gs.eye(model.dim_noise)
+    N = 10 * gs.eye(model.dim_obs)
+    initial_covs = (P0, Q, N)
+    kalman.initialise_covariances(*initial_covs)
+
+    true_state = gs.array([0, 0])
+    true_inputs = [gs.array([dt, gs.random.uniform(-1, 1)]) for _ in
+                   range(n_traj)]
+
+    true_traj, inputs, observations = create_data(
+        kalman, true_state, true_inputs, obs_freq)
+
+    initial_state = np.random.multivariate_normal(true_state, P0)
+    estimate, uncertainty = estimation(
+        kalman, initial_state, inputs, observations, obs_freq)
+
+    plt.figure()
+    plt.plot(true_traj[:, 0], label='GT')
+    plt.plot(estimate[:, 0], label='Kalman')
+    plt.plot(estimate[:, 0] + uncertainty[:, 0], color='k', linestyle=':')
+    plt.plot(
+        estimate[:, 0] - uncertainty[:, 0], color='k', linestyle=':',
+        label='3_sigma envelope')
+    plt.plot(range(obs_freq, n_traj + 1, obs_freq), observations,
+             marker='*', linestyle='', label='Observation')
+    plt.legend()
+    plt.title('1D Localization - Position')
+
+    plt.figure()
+    plt.plot(true_traj[:, 1], label='GT')
+    plt.plot(estimate[:, 1], label='Kalman')
+    plt.plot(estimate[:, 1] + uncertainty[:, 1], color='k', linestyle=':')
+    plt.plot(
+        estimate[:, 1] - uncertainty[:, 1], color='k', linestyle=':',
+        label='3_sigma envelope')
+    plt.legend()
+    plt.title('1D Localization - Speed')
+
+    model = Localization()
+    kalman = KalmanFilter(model)
+    P0 = gs.array([1., 10., 10.])
+    P0 = np.diag(P0)
+    Q = 0.001 * gs.eye(model.dim_noise)
+    N = 0.1 * gs.eye(model.dim_obs)
+    initial_covs = (P0, Q, N)
+    kalman.initialise_covariances(*initial_covs)
+
+    true_state = gs.array([0, 0, 0])
+    true_inputs = [gs.array([dt, .5, 0., 0.05]) for _ in range(n_traj)]
+
+    true_traj, inputs, observations = create_data(
+        kalman, true_state, true_inputs, obs_freq)
+
+    initial_state = np.random.multivariate_normal(true_state, P0)
+    estimate, uncertainty = estimation(
+        kalman, initial_state, inputs, observations, obs_freq)
+
+    plt.figure()
+    plt.plot(true_traj[:, 1], true_traj[:, 2], label='GT')
+    plt.plot(estimate[:, 1], estimate[:, 2], label='Kalman')
+    plt.scatter(observations[:, 0], observations[:, 1], s=2, c='k',
+                label='Observation')
+    plt.legend()
+    plt.axis('equal')
+    plt.title('2D Localization')
+
+    plt.show()
```

Comment:
the examples are supposed to be able to run from command line IIRC. Instead of show this should be savefig.

See example at:

https://github.com/geomstats/geomstats/blob/master/examples/plot_expectation_maximization_manifolds.py

Reply:
In the example you linked to, the `main` function ends with `show`. I agree for the command line part, but why is savefig a better choice then ? (Real question there, I'm simply used to work with `show` and did not really consider alternatives)

Reply:
If you don't use savefig the mapltolib backend is going to try to show the figure on a screen. These examples should be able to run from the command line without hiccups, hence better use savefig.

Reply:
Hm, so far we have some `plt.show()` in almost all examples... (We shut them down when we run the examples at test time). Why is it better to save than to show? I would say this is personal preference?